bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/helper/image/AssetDeliveryHelper.java

```diff
@@ -0,0 +1,213 @@
+    }
+
+    private static void addRotationParameter(@NotNull Map<String, Object> params, @NotNull ValueMap componentProperties) {
+        int rotate = getRotation(componentProperties);
+        if (Integer.valueOf(rotate) != null && rotate != 0) {
+            params.put(ROTATE_PARAMETER, "" + rotate);
+        }
+    }
+
+    private static void addFlipParameter(@NotNull Map<String, Object> params, @NotNull ValueMap componentProperties) {
+        String flipParameter = getFlip(componentProperties);
+        if (!StringUtils.isEmpty(flipParameter)) {
+            params.put(FLIP_PARAMETER, flipParameter);
+        }
+    }
+
+    /**
+     * Retrieves the cropping rectangle, if one is defined for the image.
+     *
+     * @param properties the image component's properties
+     * @return the cropping parameters, if one is found, {@code null} otherwise
+     */
+    private static String getCropRect(@NotNull ValueMap properties) {
+        String csv = properties.get(ImageResource.PN_IMAGE_CROP, String.class);
+        if (StringUtils.isNotEmpty(csv)) {
+            try {
+                int ratio = csv.indexOf('/');
+                if (ratio >= 0) {
+                    // skip ratio
+                    csv = csv.substring(0, ratio);
+                }
+
+                String[] coords = csv.split(",");
+                int x1 = Integer.parseInt(coords[0]);
+                int y1 = Integer.parseInt(coords[1]);
+                int x2 = Integer.parseInt(coords[2]);
+                int y2 = Integer.parseInt(coords[3]);
+                int width = x2-x1;
+                int height = y2-y1;
+                return x1 + COMMA + y1 + COMMA + width + COMMA + height;
+            } catch (RuntimeException e) {
+                LOGGER.warn(String.format("Invalid cropping rectangle %s.", csv), e);
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Retrieves the rotation angle for the image, if one is present. Typically this should be a value between 0 and 360.
+     *
+     * @param properties the image component's properties
+     * @return the rotation angle
+     */
+    private static int getRotation(@NotNull ValueMap properties) {
+        String rotationString = properties.get(ImageResource.PN_IMAGE_ROTATE, String.class);
+        if (rotationString != null) {
+            try {
+                return Integer.parseInt(rotationString);
+            } catch (NumberFormatException e) {
+                LOGGER.warn(String.format("Invalid rotation value %s.", rotationString), e);
+            }
+        }
+        return 0;
+    }
+
+    /**
+     * Retrieves the flip parameters for the image, if one is present.
+     * @param properties the image component's properties
+     * @return the rotation angle
+     */
+    private static String getFlip(@NotNull ValueMap properties) {
+        boolean flipHorizontally = properties.get(com.adobe.cq.wcm.core.components.models.Image.PN_FLIP_HORIZONTAL, Boolean.FALSE);
```

Comment:
simplify `com.adobe.cq.wcm.core.components.models.Image.PN_FLIP_HORIZONTAL` to `Image.PN_FLIP_HORIZONTAL`